modules/EPMC.py

```diff
@@ -58,39 +58,47 @@ def main(cooccurrenceFile, outputFile, local=False):
         # Reading file:
         spark.read.parquet(cooccurrenceFile)
 
+        # Casting integer pmid column to string:
+        .withColumn("pmid", pf.col('pmid').cast(StringType()))
+
+        # publication identifier is a pmid if available otherwise pmcid
+        .withColumn(
+            'publicationIdentifier',
+            pf.when(pf.col('pmid').isNull(), pf.col('pmcid'))
+            .otherwise(pf.col('pmid'))
+        )
+
         # Filtering for disease/target cooccurrences:
         .filter(
             (pf.col('type') == 'GP-DS') &  # Filter gene/protein - disease cooccurrence
-            (pf.col('isMapped') == True) &  # Filtering for mapped cooccurrences
-            (pf.col('pmid').isNotNull()) &  # Excluding publications without pmid
+            (pf.col('isMapped')) &  # Filtering for mapped cooccurrences
+            # Making sure at least the pmid or the pmcid is given:
+            (pf.col('pmid').isNotNull() | pf.col('pmcid').isNotNull()) &
             (pf.length(pf.col('text')) < 600) &  # Exclude sentences with more than 600 characters
             (pf.col('label1').isin(EXCLUDED_TARGET_TERMS) == False)  # Excluding target labels from the exclusion list
         )
+
         # Renaming columns:
         .withColumnRenamed('keywordId1', 'targetFromSourceId')
         .withColumnRenamed('keywordId2', 'diseaseFromSourceMappedId')
-        .withColumnRenamed('label1', 'targetFromSource')
-        .withColumnRenamed('label2', 'diseaseFromSource')
-
-        .withColumn('pmid_str_tmp', pf.col('pmid').cast(StringType()))
     )
 
     # Report on the number of diseases, targets and associations if loglevel == "debug" to avoid cost on computation time:
-    logging.debug(f"Number of publications: {filtered_cooccurrence_df.select(pf.col('pmid_str_tmp')).distinct().count()}")
+    logging.debug(f"Number of publications: {filtered_cooccurrence_df.select(pf.col('pmid')).distinct().count()}")
```

Comment:
Again, since some records don't have PMID but have PMCID, sounds like we should count all of them?

```suggestion
    logging.debug(f"Number of publications: {filtered_cooccurrence_df.select(pf.col('publicationIdentifier')).distinct().count()}")
```

Reply:
Make sense.